packages/web/src/store/playlist-library/helpers.ts

```diff
@@ -164,6 +164,39 @@ export const renamePlaylistFolderInLibrary = (
   }
 }
 
+/**
+ * Removes folder with given id from the library.
+ * Any playlists or temporary playlists in the deleted
+ * folder are moved out of the folder.
+ * Note that this assumes that folders cannot be nested within one another.
+ * If we enable nesting folders in the future, this function must be updated.
+ * @param library
+ * @param folderId
+ * @returns the updated playlist library
+ */
+export const removePlaylistFolderInLibrary = (
+  library: PlaylistLibrary,
+  folderId: string
+): PlaylistLibrary => {
+  if (!library.contents) return library
+  const folder = library.contents.find(item => {
+    return item.type === 'folder' && item.id === folderId
+    // Need to cast here because TS doesn't know that the result has to be a folder or undefined due to `item.type === 'folder'`
+  }) as PlaylistLibraryFolder | undefined
+  if (!folder) return library
+  const folderIndex = library.contents.findIndex(item => {
```

Comment:
Could we find the folder index first and use that to get the folder, avoiding the extra `find`?